### PR TITLE
UnitTests for PAC

### PR DESCRIPTION
**Pull request by hendrikebbers**
Refactoring of PAC
Fixes for PAC

### CODE REVIEW

File: openwebstart/src/main/java/com/openwebstart/proxy/pac/PacUtils.java [hunk 1]
```diff
@@ -28,27 +30,28 @@
...
+                                .map(sa -> new Proxy(Proxy.Type.HTTP, sa))
+                                .orElseThrow(() -> new IllegalArgumentException("HTTP Proxy must be specified with valid address"));
+                    } else if (token.startsWith(SOCKS)) {
+                        return getInetSocketAddress(token)
+                                .map(sa -> new Proxy(Proxy.Type.SOCKS, sa))
+                                .orElseThrow(() -> new IllegalArgumentException("SOCKS Proxy must be specified with valid address"));
+                    } else if (token.startsWith(DIRECT)) {
+                        return Proxy.NO_PROXY;
+                    } else {
+                        throw new IllegalArgumentException("Unrecognized proxy token: " + token);
```
Comment (sclassen) [inline openwebstart/src/main/java/com/openwebstart/proxy/pac/PacUtils.java]: This used to be a LOG.debug.
Now it throws and thus ending the processing of the stream
Could you LOG and return null and then in a later stage filter out any null values???

File: openwebstart/src/test/java/com/openwebstart/proxy/pac/PacUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        //when
+        final List<Proxy> proxiesFromPacResult = PacUtils.getProxiesFromPacResult(pacResult);
+
+        //than
+        Assertions.assertNotNull(proxiesFromPacResult);
+        Assertions.assertEquals(1, proxiesFromPacResult.size());
+        Assertions.assertEquals(Proxy.NO_PROXY, proxiesFromPacResult.get(0));
+
+    }
+}
```
Comment (sclassen) [inline openwebstart/src/test/java/com/openwebstart/proxy/pac/PacUtilsTest.java]: Could you add a test case for an invalid `pacResult`

File: openwebstart/src/main/java/com/openwebstart/proxy/pac/PacUtils.java (hunk 1 above)
Comment (hendrikebbers) [reply openwebstart/src/main/java/com/openwebstart/proxy/pac/PacUtils.java]: I would say that an error by parsing the string should end in an error. If a pac file returns a string that does not match the definition of PAC results an exception should be thrown. Why return null in such case?

File: openwebstart/src/test/java/com/openwebstart/proxy/pac/PacUtilsTest.java (hunk 2 above)
Comment (hendrikebbers) [reply openwebstart/src/test/java/com/openwebstart/proxy/pac/PacUtilsTest.java]: will do

Comment (hendrikebbers) [reply openwebstart/src/test/java/com/openwebstart/proxy/pac/PacUtilsTest.java]: done

**Pull request merged**